packages/@blockprotocol/graph/src/stdlib/subgraph/element/data-type.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  BaseUri,
+  extractBaseUri,
+  extractVersion,
+  VersionedUri,
+} from "@blockprotocol/type-system/slim";
+
+import { OntologyTypeEditionId } from "../../../types/ontology.js";
+import { DataTypeWithMetadata } from "../../../types/ontology/data-type.js";
+import { Subgraph } from "../../../types/subgraph.js";
+import { isDataTypeVertex } from "../../../types/subgraph/vertices.js";
+
+/**
+ * Returns all `DataTypeWithMetadata`s within the vertices of the subgraph
+ *
+ * @param subgraph
+ */
+export const getDataTypes = (subgraph: Subgraph): DataTypeWithMetadata[] => {
+  return Object.values(
+    Object.values(subgraph.vertices).flatMap((versionObject) =>
+      Object.values(versionObject)
+        .filter(isDataTypeVertex)
+        .map((vertex) => vertex.inner),
+    ),
+  );
+};
+
+/**
+ * Gets a `DataTypeWithMetadata` by its `VersionedUri` from within the vertices of the subgraph. Returns `undefined` if
+ * the data type couldn't be found.
+ *
+ * @param subgraph
+ * @param dataTypeId
+ * @throws if the vertex isn't a `DataTypeVertex`
+ */
+export const getDataTypeById = (
+  subgraph: Subgraph,
+  dataTypeId: VersionedUri,
+): DataTypeWithMetadata | undefined => {
+  const [baseUri, version] = [
+    extractBaseUri(dataTypeId),
+    extractVersion(dataTypeId),
+  ];
```

Comment:
Presumably at some point we can have a
```ts
const { baseUri, version } = destructureVersionedUrI(dataTypeId);
```

Reply:
Yeah we've already got that in HASH, we could look adding it here later on as an extension too